LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2Condition_NotItsOwnTurn.uc

```diff
@@ -0,0 +1,24 @@
+//---------------------------------------------------------------------------------------
+//  FILE:    X2Condition_NotItsOwnTurn.uc
+//  AUTHOR:  Grobobobo
+//  PURPOSE: Condition to not trigger an ability on the unit's turn
+//---------------------------------------------------------------------------------------
+class X2Condition_NotItsOwnTurn extends X2Condition;
+
+event name CallMeetsCondition(XComGameState_BaseObject kTarget) 
+{
+    local XComGameState_Unit UnitState;
+    
+    UnitState = XComGameState_Unit(kTarget);
+    
+    if (UnitState != none)
+    {
+        if (`TACTICALRULES.GetUnitActionTeam() != UnitState.GetTeam())
+        {
+            return 'AA_Success'; 
+        }
+    }
+    else return 'AA_NotAUnit';
+
+    return 'AA_AbilityUnavailable';
```

Comment:
Fixes for the formatting below. I also suggest merging the `if` statements into one with an `&&` to reduce the amount of nesting, but it's not particularly important in such a small bit of code.
```suggestion
	local XComGameState_Unit UnitState;

	UnitState = XComGameState_Unit(kTarget);

	if (UnitState != none)
	{
		if (`TACTICALRULES.GetUnitActionTeam() != UnitState.GetTeam())
		{
			return 'AA_Success'; 
		}
	}
	else return 'AA_NotAUnit';

	return 'AA_AbilityUnavailable';
```

Reply:
Wait, Won't doing UnitState.GetTeam() crash if UnitState is null?